source/Esf.Website/App/esfState/esfStateRunQueryCommand.service.ts

```diff
@@ -0,0 +1,78 @@
+﻿import { Injectable } from '@angular/core';
+import { EsfStateViewModel } from '../common/models/esfStateViewModel';
+import { EsfStateDto } from '../common/models/esfStateDto';
+import { EsfStateProvider } from '../common/interfaces/esfStateProvider.interface';
+import { EsfStateQueryResultConsumer } from '../common/interfaces/esfStateQueryResultConsumer.interface';
+import { EsfStateService } from './esfState.service';
+import { EsfStateValidationService } from '../common/services/esfStateValidation.service';
+import { EsfQueryRunnerService } from '../esfQueryRunner/esfQueryRunner.service';
+import { CommandState } from '../common/models/commandState';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+
+@Injectable()
+export class EsfStateRunQueryCommand {
+    private stateProvider: EsfStateProvider;
+    private queryResultStream: Subject<QueryResult>;
+    private commandStateStream: Subject<CommandState>;
+
```

Comment:
I would be merging queryResultStream and commandStateStream into a single Observable. This applies for the other command as well.